onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -368,121 +628,185 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
+
+            params.BluesteinZChirpParams.AFFT.Sizes = GetReshapedDimensions(inputDims, m_axis);
+            params.BluesteinZChirpParams.AFFT.Sizes[1] = M;
+            params.BluesteinZChirpParams.AFFT.Sizes.back() = 2;
+            Dml::GetDescendingPackedStrides(params.BluesteinZChirpParams.AFFT.Sizes, params.BluesteinZChirpParams.AFFT.Strides);
+
+            params.BluesteinZChirpParams.B.Sizes = std::array<uint32_t, 4> { 1, 1, M, 2 };
+            params.BluesteinZChirpParams.B.Strides = std::array<uint32_t, 4> { M * 2, M * 2, 2, 1 };
+
+            params.BluesteinZChirpParams.BFFT.Sizes = std::array<uint32_t, 4> { 1, 1, M, 2 };
+            params.BluesteinZChirpParams.BFFT.Strides = std::array<uint32_t, 4> { M * 2, M * 2, 2, 1 };
+
+            auto zChirpBufferByteSize = sizeof(float) * ComputeElementCountFromDimensions(params.BluesteinZChirpParams.ZChirp.Sizes);
+            auto aIntermediateBufferByteSize = sizeof(float) * ComputeElementCountFromDimensions(params.BluesteinZChirpParams.AFFT.Sizes);
+            auto bIntermediateBufferByteSize = sizeof(float) * ComputeElementCountFromDimensions(params.BluesteinZChirpParams.BFFT.Sizes);
+
+            auto& zchirp_resource = params.BluesteinZChirpParams.ZChirp.Resource;
+            auto& afft_resource = params.BluesteinZChirpParams.AFFT.Resource;
+            auto& b_resource = params.BluesteinZChirpParams.B.Resource;
+            auto& bfft_resource = params.BluesteinZChirpParams.BFFT.Resource;
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(zChirpBufferByteSize, &zchirp_resource));
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(aIntermediateBufferByteSize, &afft_resource));
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(bIntermediateBufferByteSize, &b_resource));
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(bIntermediateBufferByteSize, &bfft_resource));
+
+            // The AFFT call takes input A, and produces output A_FFT.
+            //
+            // Input A: This is a pow-2 padded and chirp-weighted representation of the signal represented by "inputResource"
+            // Therefore the dftLength is not correct for AFFT, it should be NextPowerOf2(2*dftLength-1)
+            //
+            // The weighted representation should be calculated by passing in the chirp to the dft (like a window function).
+            // Padding should be handled by the shader.
+            PrepareStockhamFFTParams(context,
+                                     inputResource, inputDims,
+                                     afft_resource.Get(), params.BluesteinZChirpParams.AFFT.Sizes,
+                                     M,
+                                     m_axis,
+                                     1,
+                                     params.BluesteinZChirpParams.AFFTParams);
+            params.BluesteinZChirpParams.AFFTParams.Window = params.BluesteinZChirpParams.ZChirp;
```

Comment:
```suggestion
            PrepareStockhamFFTParams(
                context,
                inputResource, inputDims,
                afft_resource.Get(), params.BluesteinZChirpParams.AFFT.Sizes,
                M,
                m_axis,
                1,
                params.BluesteinZChirpParams.AFFTParams
            );
```

[minor] I quite like the way you indented here...

```
        barriers[1] = CD3DX12_RESOURCE_BARRIER::Transition(
            outputResource,
            D3D12_RESOURCE_STATE_COMMON,
            D3D12_RESOURCE_STATE_UNORDERED_ACCESS
        );
```

...which makes it easier to see parameter additions/reorderings (bonus: and then a search&replace for changing function names later doesn't shift everything out of alignment).